### DISCUSSION TITLE
Отобразить строки с одинаковым значением

### DISCUSSION THREAD

**Discussion by MagosDominusWyrd** [totumonline/totum-ru-issues-and-discussions#278] [Как это сделать?] [answered]
Добрый день. Не могу никак решить задачку по созданию простейшего модуля показа дубликатов.

Дано:
Таблица, в одном столбце которой могут попадаться одинаковые значения

Нужно:
По нажатию кнопки отображать все строки через LinkToTable на эту же таблицу с префильтром, в которых есть одинаковые значения, т.е., например у нас есть 6 строк, в 3 из которых значение 0, в двух значение 1. в одной значение 2. Нужно отобразить все строки со значениями 0 и 1.

Как пробовал реализовать:
Опуская мелкие детали (создал кнопку, создал префильтр для последующего LinkToTable и т.п.),
1 - Забирал список всех значений из нужного столбца;
2 - применял к нему listUniq;
3 - применял к первому списку listMinus списка, полученного через listUniq;
4 - ничего дельного не получалось =)

Подскажите, пожалуйста, как это можно реализовать =)

**Comment by erpic-ru** [comment, marked as answer]
Добрый день! Я бы сделал проще:

1. Добавить техническое поле для создателя с чекбоксом double, с кодом при добавлении/изменении, что если есть такое значение то ставить true
2. Префильтр на него 
3. Возможно, ещё код-действия recalculate понадобится

**Reply by MagosDominusWyrd** [reply]
Спасибо за идею, это действительно проще =)

**Reply by ttm-alexeypolunin** [reply]
В текущей версии Тотум действительно нет запроса, позволяющего получить список дубликатов быстрым запросом к БД.

Те сейчас через запрос можно взять все и потом пройтись по нему listReplace заменяя на пустые те, которые не дублируются и потом отфильтровать оставшиеся. Но если таблица большая или будет большая 1М строк например (может и меньше) — это ляжет.

В PRO имея мидл-скилл в SQL можно напрямую написать запрос к БД на получение списка дубликатов. Как, описывать не буду, что бы те, кто этого скила не имеет не положили себе насмерть базу — «в атомной подводной лодке не во все можно стрелять из пистолета»

Эта потребность известна. У нас в списке разработки записан набор функций, которые смогут получать обработанные штуки через БД типа:

— сумма по колонке
— количество строк
— среднее
— количество уникальных
— только уникальные
— только дубли
— итп
